Lib/http/server.py

```diff
@@ -736,6 +750,10 @@ def send_head(self):
                 new_url = urllib.parse.urlunsplit(new_parts)
                 self.send_header("Location", new_url)
                 self.send_header("Content-Length", "0")
+                # User specified response_headers
+                if self.response_headers is not None:
+                    for header, value in self.response_headers.items():
+                        self.send_header(header, value)
```

Comment:
let's make it a private method, say `self._add_custom_response_headers` or something like that